plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/TestBridgingHiveMetastore.java

```diff
@@ -47,4 +55,31 @@ void afterAll()
     {
         hiveHadoop.stop();
     }
+
+    @Test
+    void testDropTable()
+    {
+        String databaseName = "test_database_" + randomNameSuffix();
+        Database.Builder database = Database.builder()
+                .setDatabaseName(databaseName)
+                .setOwnerName(Optional.empty())
+                .setOwnerType(Optional.empty());
+        metastore.createDatabase(database.build());
+
+        String tableName = "test_table" + randomNameSuffix();
+        Table.Builder table = Table.builder()
+                .setDatabaseName(databaseName)
+                .setTableName(tableName)
+                .setTableType(EXTERNAL_TABLE.name())
+                .setOwner(Optional.empty());
+        table.getStorageBuilder()
+                .setStorageFormat(fromHiveStorageFormat(PARQUET));
+        metastore.createTable(table.build(), NO_PRIVILEGES);
+
+        metastore.dropTable(databaseName, tableName, false);
+        // Simulate retry in case of timout. This call should succeed
```

Comment:
Check that the table that we're trying drop again does not actually exist anymore. (`!metastore.getTables().contains...` / `metastore.getTable(...)`)